bndtools.core.test/src/bndtools/core/test/editors/quickfix/BuildpathQuickFixProcessorTest.java

```diff
@@ -826,8 +853,25 @@ void withFQClassLiteral_asAnnotationParameter_suggestsBundles(SoftAssertions sof
 			suggestsBundle("bndtools.core.test.fodder.iface", "1.0.0", "iface.bundle.MyInterface"));
 	}
 
+	@Disabled("Can't seem to make this work - binding information seems to be missing from class literal, I think this is"
```

Comment:
If you need parameter names, then you need to tell the compiler (javac and eclipse) to compile the parameter names in. So you probably need to make a change to this test project similar to:

https://github.com/bndtools/bnd/commit/26f649a722fcd0dcf3d334972e887a6453e9ce4f

Reply:
Thanks - but I don't think that will help. The quick fix processor doesn't need to know the variable names, only their types.

It seemed that even though `ClassWithInterfaceExtendingMyInterface` is on the classpath and I have bindings recovery turned on, Eclipse's compiler wasn't properly generating the type binding information for the hierarchy of `ClassWithInterfaceExtendingMyInterface` so that I could find the missing type.

I got to the point where I felt that this was actually a bug in Eclipse that it wasn't fully recovering the incomplete type. However, I just had another idea that I will retry.

Reply:
I retried the idea and re-confirmed my original diagnosis. This time I documented the issue more precisely in the test code itself. I left in there in case the Eclipse issue is ever resolved, then this test case can be re-instated.

I'm able to get the binding information for `ClassWithInterfaceExtendingMyInterface`. However, when I call `getInterfaces()` on the type binding it returns an empty array. What it *should* be doing is returning a recovered type binding for its implemented interface `InterfaceExtendingMyInterface` (which the quick fix processor would then know how to fix).

The annoying thing is that if you print out the type binding for `ClassWithInterfaceExtendingMyInterface` or examine its private state variables, you can see that Eclipse knows it implements `InterfaceExtendingMyInterface` and has a recovered binding for it. It's just not returning it in `getInterfaces()`.

Reply:
Hmm. Is there an Eclipse bug filed for this, I wonder?

Reply:
There is now! https://bugs.eclipse.org/bugs/show_bug.cgi?id=566145